rules/PoiGeneric.js

```diff
@@ -164,8 +165,17 @@ function additiveScore(map, e1, e2) {
 
     var reason = result.reasons;
 
-    var searchRadius = Math.max(exports.getSearchRadius(e1), 
-        exports.getSearchRadius(e2));
+    var t1 = e1.getTags().toDict();
+    var t2 = e2.getTags().toDict();
+
+    var searchRadius = Math.max(exports.getSearchRadius(e1), exports.getSearchRadius(e2));
+
+    var stopPositionThreshold = 0.7;
+    if (getRelatedTagsByThreshold("public_transport=stop_position", e1.getTags().toDict(), stopPositionThreshold).length > 0 &&
+        getRelatedTagsByThreshold("public_transport=stop_position", e2.getTags().toDict(), stopPositionThreshold).length > 0)
+    {
+      searchRadius = Math.min(exports.getSearchRadius(e1), exports.getSearchRadius(e2));
```

Comment:
What happens if you make all tags use the min of the two search radii instead of a special rule for stop_positions?